app/controllers/products_controller.rb

```diff
@@ -0,0 +1,58 @@
+class ProductsController < ApplicationController
+  def index
+    @vendor = Vendor.find(params[:vendor_id])
+    @products = @vendor.products
+  end
+
+  def new
+    @product = Product.new
+    @vendor = Vendor.find(params[:vendor_id])
+  end
+
+  def create
+    @vendor = Vendor.find(params[:vendor_id])
+    product = Product.new(create_params[:product])
+
+    if product.valid?
+      product.save
+
+      redirect_to "/vendors/#{ @vendor.id }/products"
```

Comment:
[Named routes](http://api.rubyonrails.org/classes/ActionDispatch/Routing.html) might be nice here.